src/NServiceBus.Metrics/MetricsOptions.cs

```diff
@@ -65,14 +65,16 @@ public void EnableCustomReport(Func<string, Task> func, TimeSpan interval)
         /// </summary>
         /// <param name="serviceControlMetricsAddress">The transport address of the ServiceControl instance</param>
         /// <param name="interval">Interval between consecutive reports</param>
+        /// <param name="instanceId">Unique identifier of endpoint physical instance</param>
         [ObsoleteEx(Message = "Not for public use.")]
-        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval)
+        public void SendMetricDataToServiceControl(string serviceControlMetricsAddress, TimeSpan interval, string instanceId)
```

Comment:
Why not

- If identifier is set in core, use that
- Otherwise, figure out what the machine name is and display a warning in log that they might want to use identifier

The fact that we're already putting in a mandatory `string.empty` doesn't seem right. And we can now set identifier in two places. If people are already setting host identifier and it's still mandatory here, that's pretty confusing.